src/main/java/net/syscon/elite/service/PrisonToPrisonMoveSchedulingService.java

```diff
@@ -154,29 +154,34 @@ private OffenderIndividualSchedule scheduleMove(final OffenderBooking booking,
     @VerifyBookingAccess
     @HasWriteScope
     public void cancel(final Long bookingId, final Long scheduledMoveId, final String transferCancellationReasonCode) {
-        final var scheduledMove = scheduleRepository.findById(scheduledMoveId).orElseThrow(() -> EntityNotFoundException.withMessage("Scheduled prison move with id %s not found.", scheduledMoveId));
+        final var move = scheduleRepository.findById(scheduledMoveId).orElseThrow(() -> EntityNotFoundException.withMessage("Scheduled prison move with id %s not found.", scheduledMoveId));
 
-        final var transferCancellationReason = transferCancellationReasonRepository.findById(TransferCancellationReason.pk(transferCancellationReasonCode)).orElseThrow(() -> EntityNotFoundException.withMessage("Cancellation reason %s not found.", transferCancellationReasonCode));
+        checkIsPrison(move);
+        checkIsAssociated(bookingId, move);
+        checkIsActive(move.getOffenderBooking());
+        checkCanCancel(move);
 
-        checkAssociated(bookingId, scheduledMove);
+        move.setEventStatus(eventStatusRepository.findById(EventStatus.CANCELLED).orElseThrow(() -> EntityNotFoundException.withMessage("Event status cancelled not found.")));
+        move.setCancellationReason(transferCancellationReasonRepository.findById(TransferCancellationReason.pk(transferCancellationReasonCode)).orElseThrow(() -> EntityNotFoundException.withMessage("Cancellation reason %s not found.", transferCancellationReasonCode)));
 
-        checkArgument(scheduledMove.getOffenderBooking().isActive(), "Booking with id %s is not active.", bookingId);
+        scheduleRepository.save(move);
 
-        checkCanCancel(scheduledMove);
-
-        scheduledMove.setEventStatus(eventStatusRepository.findById(EventStatus.CANCELLED).orElseThrow(() -> EntityNotFoundException.withMessage("Event status cancelled not found.")));
-        scheduledMove.setCancellationReason(transferCancellationReason);
+        log.debug("Cancelled scheduled prison to prison move with id {} for offender {}", move.getId(), move.getOffenderBooking().getOffender().getNomsId());
+    }
 
-        scheduleRepository.save(scheduledMove);
+    private void checkIsActive(final OffenderBooking booking) {
+        checkArgument(booking.isActive(), "Offender booking for prison to prison move with id %s is not active.", booking.getBookingId());
+    }
 
-        log.debug("Cancelled scheduled prison to prison move with id {} for offender {}", scheduledMove.getId(), scheduledMove.getOffenderBooking().getOffender().getNomsId());
+    private void checkIsPrison(final OffenderIndividualSchedule scheduledMove) {
+        checkArgument(PRISON_TRANSFER.equals(scheduledMove.getEventType()), "Scheduled move with id %s not a prison move.", scheduledMove.getId());
```

Comment:
This is the key thing here to check/confirm.  I believe this is correct in terms of identification purposes as to whether this is a prison move or not.